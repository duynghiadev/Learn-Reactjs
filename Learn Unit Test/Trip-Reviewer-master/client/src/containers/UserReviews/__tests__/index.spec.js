import React from 'react';
import configureStore from 'redux-mock-store';
import { create } from 'react-test-renderer';

import UserPosts from '..';

jest.mock('react-router-dom', () => ({ Link: 'Link' }));
jest.mock('./../../../components/Common/EmptyViewAddTrip', () => 'EmptyView');
jest.mock('./../../../actions', () => ({
  getUserReviews: jest.fn(() => ({
    type: 'getTripWithReviewer',
  })),
}));

const mockStore = configureStore();

const mockComponent = (initialState = {}, props) => {
  const store = mockStore(initialState);

  return <UserPosts {...props} store={store} />;
};

describe('<UserPosts />', () => {
  it('should render component with login and no posts', () => {
    const initialState = {
      users: {
        login: {
          id: 'id',
        },
        userPosts: [],
      },
    };
    const tree = create(mockComponent(initialState)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with login and several posts', () => {
    const initialState = {
      users: {
        login: {
          id: 'id',
        },
        userPosts: [
          {
            _id: 'id1',
            title: 'title1',
            country: 'country1',
            createdAt: 'createdAt1',
          },
          {
            _id: 'id2',
            title: 'title2',
            country: 'country2',
            createdAt: 'createdAt2',
          },
        ],
      },
    };
    const tree = create(mockComponent(initialState)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with no login', () => {
    const initialState = {
      users: {},
    };
    const tree = create(mockComponent(initialState)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
