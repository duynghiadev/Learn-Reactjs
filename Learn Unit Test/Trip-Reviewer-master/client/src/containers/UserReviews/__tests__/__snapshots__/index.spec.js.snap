// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserPosts /> should render component with login and no posts 1`] = `
<div
  className="user-posts-container limited-width"
>
  <h2>
    My stories
  </h2>
  <EmptyView
    buttonText="Add my first story"
    message="You do not have any story yet."
  />
</div>
`;

exports[`<UserPosts /> should render component with login and several posts 1`] = `
<div
  className="user-posts-container limited-width"
>
  <h2>
    My stories
  </h2>
  <table
    className="reviews-table"
  >
    <thead>
      <tr>
        <th>
          Title
        </th>
        <th
          className="only-wide-screen"
        >
          Country
        </th>
        <th>
          Date
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="post-title"
        >
          <Link
            to="/user/edit-post/id1"
          >
            title1
          </Link>
        </td>
        <td
          className="only-wide-screen"
        >
          country1
        </td>
        <td>
          Invalid Date
        </td>
      </tr>
      <tr>
        <td
          className="post-title"
        >
          <Link
            to="/user/edit-post/id2"
          >
            title2
          </Link>
        </td>
        <td
          className="only-wide-screen"
        >
          country2
        </td>
        <td>
          Invalid Date
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<UserPosts /> should render component with no login 1`] = `
<div
  className="user-posts-container limited-width"
>
  <h2>
    My stories
  </h2>
</div>
`;
